peps/pep-0797.rst

```diff
@@ -0,0 +1,271 @@
+        with open("spanish_inquisition.txt") as unshareable:
+            interp = interpreters.create()
+            with interpreters.share(unshareable) as proxy:
+                interp.prepare_main(file=proxy)
+                interp.exec("file.write('I didn't expect the Spanish Inquisition')")
+
+
+    .. note::
+
+        ``None`` cannot be used with this function, as ``None`` is a special
+        value reserved for dead object proxies. Since ``None`` is natively
+        shareable, there's no need to pass it to this function anyway.
+
+.. function:: concurrent.interpreters.share_forever(obj)
+
+    Similar to :func:`~concurrent.interpreters.share`, but *does not* give the resulting
+    proxy a context, meaning it will live forever (unless a call to ``share``
+    explicitly gives the proxy a new lifetime). As such, this function does not
+    return a :term:`context manager`.
+
+    For example:
+
+    .. code-block:: python
+
+        from concurrent import interpreters
+
+        with open("spanish_inquisition.txt") as unshareable:
+            interp = interpreters.create()
+            proxy = interpreters.share_forever(unshareable)
+            interp.prepare_main(file=proxy)
+            # Note: the bound method object for file.write() will also live
+            # forever in a proxy.
+            interp.exec("file.write('I didn't expect the Spanish Inquisition')")
+
+    .. warning::
+
+        Proxies created as a result of the returned proxy (for example, bound
+        method objects) will also exist for the lifetime of the interpreter,
+        which can lead to high memory usage.
+
+
+Multithreaded Scaling
+---------------------
+
+Since an object proxy mostly interacts with an object normally, there shouldn't
+be much additional overhead on using the object once the thread state has been
+switched. However, this means that when the :term:`GIL` is enabled, you may lose
+some of the concurrency benefits from subinterpreters, because threads will be
+stuck waiting on the GIL of a wrapped object's interpreter.
+
+Backwards Compatibility
+=======================
+
+In order to implement the immortality mechanism used by shared object proxies,
+several assumptions had to be made about the object lifecycle in the C API.
+So, some best practices in the C API (such as using the object allocator for
+objects) are made harder requirements by the implementation of this PEP.
+
+The author of this PEP believes it is unlikely that this will cause breakage,
+as he has not ever seen code in the wild that violates the assumptions made
+about the object lifecycle as required by the reference implementation.
+
+Security Implications
+=====================
+
+The largest issue with shared object proxies is that in order to have
+thread-safe reference counting operations, they must be :term:`immortal`,
+which prevents any concurrent modification to their reference count.
+This can cause them to take up very large amounts of memory if mismanaged.
```

Comment:
Large amounts of memory -> denial of service attack, which is the actual security concern here